### PR TITLE
Move files and change names

### PR DESCRIPTION
**Pull request by adithyaov**
- Move generated files under `Internal.Generated`
- Rename `decomposeChar` to `decompose`
- Rename functions in `Properties.Core` to match the names in `Data.Char`

### CODE REVIEW

File: exe/Parser/Text.hs [hunk 1]
```diff
@@ -529,32 +529,32 @@ genModules indir outdir props = do
           & Stream.fold (Fold.mkPureId (++) [])
 
     let compositions =
-            ( "Unicode.Properties.Compose"
+            ( "Unicode.Internal.Generated.UnicodeData.Compositions"
```
Comment (harendra-kumar) [inline exe/Parser/Text.hs:532]: Why do we need `UnicodeData`? Can we avoid that to keep the name shorter?

File: unicode-data.cabal [hunk 2]
```diff
@@ -64,19 +64,25 @@ library
   import: default-extensions, compile-options
   default-language: Haskell2010
   exposed-modules:
+      Unicode.Properties.Core
       Unicode.Properties.CombiningClass
       Unicode.Properties.Compose
       Unicode.Properties.Decompose
       Unicode.Properties.DecomposeHangul
```
Comment (harendra-kumar) [inline unicode-data.cabal:71]: Are we going to have other modules under `Unicode` other than `Properties`? Should we consider removing it to keep the module names shorter?

Comment (harendra-kumar) [reply unicode-data.cabal:71]: Any thoughts on `DecomposeHangul` vs `Decompose.Hangul`?

File: exe/Parser/Text.hs (hunk 1 above)
Comment (adithyaov) [reply exe/Parser/Text.hs:532]: It's because this module is `Generated` from `UnicodeData.txt`

File: unicode-data.cabal (hunk 2 above)
Comment (adithyaov) [reply unicode-data.cabal:71]: > Are we going to have other modules under Unicode other than Properties? Should we consider removing it to keep the module names shorter?

Yes, we can remove `Properties`. It's implied anyway.

Comment (adithyaov) [reply unicode-data.cabal:71]: > Any thoughts on DecomposeHangul vs Decompose.Hangul?

I think a proper thing to do is to move `decomposeCharHangul` into `Decompose` and rename `DecomposeHangul` to just `Hangul`.

Comment (adithyaov) [reply unicode-data.cabal:71]: I'll do both of these in another PR.

File: exe/Parser/Text.hs (hunk 1 above)
Comment (adithyaov) [reply exe/Parser/Text.hs:532]: We should remove it though. It shouldn't really matter where it's generated from anyway.

File: unicode-data.cabal (hunk 2 above)
Comment (adithyaov) [reply unicode-data.cabal:71]: See #17 and #18

File: exe/Parser/Text.hs (hunk 1 above)
Comment (adithyaov) [reply exe/Parser/Text.hs:532]: Just like `Generated.DerivedCoreProperties` is generated from `DerivedCoreProperties.txt`.

Comment (adithyaov) [reply exe/Parser/Text.hs:532]: These are generated modules anyway and the users should not directly use these.

Comment (harendra-kumar) [reply exe/Parser/Text.hs:532]: Can we name the source of the file in the generated files e.g.:

```
 -- autogenerated from ucd/DerivedCoreProperties.txt 
 ```

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**